cachito/web/api_v1.py

```diff
@@ -211,15 +212,24 @@ def download_archive(request_id):
         )
         raise InternalServerError()
 
+    checksum = compute_file_checksum(get_hasher(), str(bundle_dir.bundle_archive_file))
+    store_checksum = bundle_dir.bundle_archive_checksum.read_text(encoding="utf-8")
+    if checksum != store_checksum:
+        msg = f"Checksum of bundle archive {bundle_dir.bundle_archive_file} has been changed."
+        flask.current_app.logger.error(msg)
+        raise InternalServerError(msg)
+
     flask.current_app.logger.debug(
         "Sending the bundle at %s for request %d", bundle_dir.bundle_archive_file, request_id
     )
-    return flask.send_file(
+    resp = flask.send_file(
         str(bundle_dir.bundle_archive_file),
         mimetype="application/gzip",
         as_attachment=True,
         attachment_filename=f"cachito-{request_id}.tar.gz",
     )
+    resp.headers["Digest"] = f"sha-256={checksum}"
```

Comment:
According to https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Digest#directives, the checksum should be base64 encoded